### PR TITLE
[Heartbeat] Better document geo options

### PR DESCRIPTION
**Pull request by andrewvc**
This adds geo options to the `add_observer_metadata` example in the default config.
It also adds a small docs page pointing people to its use in the main heartbeat docs.

### CODE REVIEW

File: heartbeat/scripts/post_process_config.py [hunk 1]
```diff
@@ -24,7 +24,13 @@
             if section_name == "Processors":
                 output += line  # include section name in output
                 output += "processors:\n"
-                output += "  - add_observer_metadata: ~\n"
+                output += "  - add_observer_metadata: \n"
+                output += "      # Optional, but recommended geo settings for the location heartbeat is running in \n"
+                output += "      # geo: \n"
```
Comment (ruflin) [inline heartbeat/scripts/post_process_config.py]: We normally leave out the space in front of config options to differentiate from comments and make uncommenting easier. Same below.

File: heartbeat/docs/heartbeat-observer-options.asciidoc [hunk 2]
```diff
@@ -0,0 +1,5 @@
+[[configuration-observer-options]]
+== Specify Observer and Geo Options
+
+It is often useful to report the geographic location from which the check was performed, as well as other information about the heartbeat instance from which the check was performed. This data is attached to events via the <<add-observer-metadata,`add_observer_metadata`>> processor. Please see the documentation for this processor for more information on its use.
```
Comment (justinkambic) [inline heartbeat/docs/heartbeat-observer-options.asciidoc]: > It is often useful to report the geographic location from which the check was performed, as well as other information about the heartbeat instance from which the check was performed.

Perhaps we can change this to read:

> It is often useful to report geographic location and other information about the heartbeat instance from which the check was performed.


My main thought is reducing the text so it doesn't have the repetition.

**Comment by andrewvc** [comment]
@ruflin @justinkambic I've rewritten the docs and updated the indentation (I also ran `make update`). How's it look now?

File: heartbeat/docs/heartbeat-observer-options.asciidoc [hunk 3]
```diff
@@ -0,0 +1,7 @@
+[[configuration-observer-options]]
+== Specify Observer and Geo Options
+
+It is often useful to view and query various attributes of the instance heartbeat is running on. This data is attached to events via the <<add-observer-metadata,`add_observer_metadata`>> processor. This processor populates the ECS `observer.*` fields. One field of particular utility is the `observer.geo.name` field, which you can configure via this processor. Use this field to create distinctive geographic regions for your uptime checks. You might label your regions by datacenter name or geographic region, e.g. `virginia-dc-1`, or `us-east-1a`, or `virginia-us`.
```
Comment (dedemorton) [inline heartbeat/docs/heartbeat-observer-options.asciidoc]:
```suggestion
It is often useful to view and query various attributes of the instance Heartbeat is running on. This data is attached to events via the <<add-observer-metadata,`add_observer_metadata`>> processor. This processor populates the ECS `observer.*` fields. One field of particular utility is the `observer.geo.name` field, which you can configure via this processor. Use this field to create distinctive geographic regions for your uptime checks. You might label your regions by datacenter name or geographic region, e.g. `virginia-dc-1`, or `us-east-1a`, or `virginia-us`.
```

Comment (dedemorton) [reply heartbeat/docs/heartbeat-observer-options.asciidoc]: Hmmm... only change here was heartbeat > Heartbeat. I don't think GitHub handles long line comparisons very well (a good reason to consider adding line breaks).

File: heartbeat/docs/heartbeat-observer-options.asciidoc [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+
+Note that heartbeat should not be used with the similarly named <<add-host-metadata,`add_host_metadata`>> processor. In ECS parlance the host is thing that is monitored, not the thing doing the monitoring, which is the observer.
```
Comment (dedemorton) [inline heartbeat/docs/heartbeat-observer-options.asciidoc]:
```suggestion
Note that Heartbeat should not be used with the similarly named <<add-host-metadata,`add_host_metadata`>> processor. In ECS parlance the host is the thing that is monitored, not the thing doing the monitoring, which is the observer.
```

File: heartbeat/heartbeat.yml [hunk 5]
```diff
@@ -128,7 +128,13 @@ output.elasticsearch:
 
 #================================ Processors =====================================
 processors:
-  - add_observer_metadata: ~
+  - add_observer_metadata: 
+# Optional, but recommended geo settings for the location heartbeat is running in 
+#    geo:
```
Comment (ruflin) [inline heartbeat/heartbeat.yml]: Hm, that works too. The more standard way we did it so far is to have `#geo`, see lines above. No strong opinion but we should consistent I think.

**Comment by andrewvc** [comment]
@ruflin better now? I could use an LGTM here.

File: heartbeat/heartbeat.yml [hunk 6]
```diff
@@ -128,8 +128,15 @@ output.elasticsearch:
 
 #================================ Processors =====================================
 processors:
-  - add_observer_metadata: ~
+  - add_observer_metadata: 
+    # Optional, but recommended geo settings for the location heartbeat is running in 
+    # geo:
```
Comment (ruflin) [inline heartbeat/heartbeat.yml]:
```suggestion
    #geo: 
```
Sorry, still not happy with the indentation. I'm mainly worried that users uncommenting entries will mess things up.

File: heartbeat/heartbeat.yml [hunk 7]
```diff
@@ -128,8 +128,15 @@ output.elasticsearch:
 
 #================================ Processors =====================================
 processors:
-  - add_observer_metadata: ~
+  - add_observer_metadata: 
+    # Optional, but recommended geo settings for the location heartbeat is running in 
+    # geo: 
 
+    # Token describing this location
+    #  name: us-east-1a
```
Comment (ruflin) [inline heartbeat/heartbeat.yml]:
```suggestion
      #name: us-east-1a
```

File: heartbeat/heartbeat.yml [hunk 8]
```diff
@@ -128,8 +128,15 @@ output.elasticsearch:
 
 #================================ Processors =====================================
 processors:
-  - add_observer_metadata: ~
+  - add_observer_metadata: 
+    # Optional, but recommended geo settings for the location heartbeat is running in 
+    # geo: 
 
+    # Token describing this location
```
Comment (ruflin) [inline heartbeat/heartbeat.yml]:
```suggestion
      # Token describing this location
```

File: heartbeat/heartbeat.yml [hunk 9]
```diff
@@ -128,8 +128,15 @@ output.elasticsearch:
...
+    #  name: us-east-1a
+
+    #  Lat, Lon in decimal
```
Comment (ruflin) [inline heartbeat/heartbeat.yml]:
```suggestion
      # Lat, Lon in decimal
```

File: heartbeat/heartbeat.yml [hunk 10]
```diff
@@ -128,8 +128,15 @@ output.elasticsearch:
...
+    #  location: "37.926868, -78.024902"
```
Comment (ruflin) [inline heartbeat/heartbeat.yml]:
```suggestion
      #location: "37.926868, -78.024902"
```

**Comment by ruflin** [comment]
Change in the config now looks good to me, but I think you need to update the script and then run `make update` to get it green.

**Comment by andrewvc** [comment]
@ruflin just pushed. Can I get an LGTM WFG?

**Comment by ruflin** [comment]
@andrewvc Now that you run `make update` you reverted the changes you made before :-(

**Pull request merged**